addons/mail/models/mail_message.py

```diff
@@ -689,6 +689,8 @@ def _update_content(self, body, attachment_ids):
             }
             attachement_values = thread._message_post_process_attachments([], attachment_ids, message_values)
             self.update(attachement_values)
+        if self.starred:
+            self.toggle_message_starred()
```

Comment:
The intent of this method is to make any content update on the message.
Message deletion is done by means of updating to an empty body.
There should be a condition on body becoming empty.
It would be surprising that any content update on message (including non-empty body) always un-star the message.

These LOCs will still be bit strange by themselves, but it's a good fix.
Could be nice to put a brief comment just to clarify what's the deal with updating body on starred message implying toggling starred.
Something like _"this message is basically deleted, hence should be un-starred"_.

Reply:
Note there are a lot more criteria than `body` being empty to consider the message empty, see `_computeIsEmpty`.

Reply:
@alexkuhn I have added a comment to get better understanding about this logic.

Reply:
That's a bit strange modeling. An empty message has nothing to do with star status. Keep conditions localized in your method and avoid adding complexity on model for a corner case.

By the way you should not talk about deletion when body is empty, that's counter intuitive in Odoo framework.